src/main.ts

```diff
@@ -0,0 +1,63 @@
+import express from 'express'
+import { createServer } from 'http'
+import { Server } from 'socket.io'
+import { Person, PersonSendImage, PersonSendMessage } from './types'
+
+const app = express()
+const server = createServer(app)
+const io = new Server(server, {
+  cors: {
+    origin: 'http://localhost:3001',
+    methods: ['GET', 'POST']
+  }
+})
+
+let userData: Array<Person> = []
+
+io.on('connection', socket => {
+  socket.on('userData', (user: Person) => {
+    userData = userData.concat(user)
+
+    socket.join(user.roomName)
+
+    socket.emit('roomMessage', `Youy Joined to ${user.roomName}`)
+
+    socket.broadcast.to(user.roomName).emit('roomMessage', `${user.userName} Joined to ${user.roomName}`)
+  })
+
+  socket.on('chatMessage', (message: PersonSendMessage) => {
+    const findUser= userData.find(user => user.clientId === socket.id)
+
+    io.to(findUser?.roomName!).emit('message', {
```

Comment:
if would be more readable, I would even say that without if this emit doesnt make any sense